cdap-ui/app/hydrator/services/plugin-config-factory.js

```diff
@@ -276,8 +279,99 @@ class HydratorPlusPlusPluginConfigFactory {
     }
     return groupConfig;
   }
+
+  validatePluginProperties(nodeInfo, errorCb) {
+    // for post-run plugins, use nodeInfo. For other plugins, get plugin property. 
+    const pluginInfo = nodeInfo.plugin? nodeInfo.plugin : nodeInfo;
+    const plugin = angular.copy(pluginInfo);
+    if (!plugin.type) {
+      plugin.type = nodeInfo.type;
+    }
+    const requestBody = {
+      stage: {
+        name: this.myHelpers.objectQuery(pluginInfo, 'name'),
+        plugin
+      }, 
+      inputSchemas: !nodeInfo.inputSchema ? [] : nodeInfo.inputSchema.map(input => {
+        let schema;
+        try {
+          schema = JSON.parse(input.schema);
+        } catch (e) {
+          // no-op
+        }
+      })
+    };
+
+    const parseResSchema = (res) => {
+      if (res.name && res.type && res.fields) {
+        return [this.getOutputSchemaObj(res)];
+      }
+      let schemaArr = [];
+      angular.forEach(res, (value, key) => {
+        if (value.name && value.type && value.fields) {
+          schemaArr.push(this.getOutputSchemaObj(value, key));
+        }
+      });
+      let recordSchemas = schemaArr.filter(schema => schema.name.substring(0, 6) === 'record');
+      let schemaArrWithoutRecordSchemas = _.difference(schemaArr, recordSchemas);
+      let schemaArrWithSortedRecordSchemas = schemaArrWithoutRecordSchemas.concat(_.sortBy(recordSchemas, 'name'));
+      return schemaArrWithSortedRecordSchemas;
+    };
+
+    const params = {
+      context: this.$state.params.namespace || this.$state.params.nsadmin
+    };
+
+    this.myPipelineApi.validateStage(params, requestBody)
+      .$promise
+      .then((res) => {
+        let errorCount; 
+        this.validating = false;
+        if (res.failures.length > 0) {
+          const { propertyErrors, inputSchemaErrors,outputSchemaErrors } = this.configurationGroupUtilities.constructErrors(res.failures);
+          errorCount = this.configurationGroupUtilities.countErrors(propertyErrors, inputSchemaErrors, outputSchemaErrors);
+          this.myAlertOnValium.show({
+            type: 'danger',
+            content: `${errorCount} error${errorCount.length > 1 ? 's': ''} found.`
```

Comment:
please change `errorCount.length` to `errorCount`. This is a bug in the existing code.